### PR TITLE
update for docker-compose-file

### PR DESCRIPTION
**Pull request by ned1313**
### Solution
Updating the docker-compose-file doc 

### Context
Doc updated to new style with more clarity
Moving the installation files to gloo/install/docker-compose-file

Note: The docker-compose-file solution does not work with the latest version of Gloo. The actual exercise files need to be updated. I'll work with @ilackarms to get v1.2.4 working properly.

### CODE REVIEW

File: docs/content/installation/gateway/docker-compose-file/_index.md [hunk 1]
```diff
@@ -4,15 +4,160 @@ weight: 5
 description: How to run Gloo Locally using Docker-Compose
 ---
 
-1. Clone the solo-docs repository and cd to this example: `git clone https://github.com/solo-io/solo-docs && cd solo-docs/gloo/docs/installation/gateway/docker-compose-file`
-1. Run `./prepare-directories.sh`
-1. You can optionally set `GLOO_VERSION` environment variable to the Gloo version you want (defaults to "0.17.4").
-1. Run `docker-compose up`
+While Gloo is typically run on Kubernetes, it doesn't need to be! You can run Gloo using Docker Compose on your local machine.
```
Comment (bettyjunod) [inline docs/content/installation/gateway/docker-compose-file/_index.md]: You can run Gloo on your local machine using Docker Compose

^ updated as it read a little awkward for me the other way.

File: docs/content/installation/gateway/docker-compose-file/_index.md [hunk 2]
```diff
@@ -4,15 +4,160 @@ weight: 5
...
 
-## Example
+Kubernetes provides APIs for config storage ([CRDs](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)), credential storage ([Secrets](https://kubernetes.io/docs/concepts/configuration/secret/)), and service discovery ([Services](https://kubernetes.io/docs/concepts/services-networking/service/)). These APIs need to be substituted with another option when Gloo is not running on Kubernetes.
 
-This configuration comes pre-loaded with an example upstream that includes an optional function service specification `serviceSpec`
-that Gloo uses to allow function level routing.
+Fortunately, Gloo provides alternate mechanisms for configuration, credential storage, and service discovery that do not require Kubernetes, including the use of local `.yaml` files, [Consul Key-Value storage](https://www.consul.io/api/kv.html) and [Vault Key-Value storage](https://www.vaultproject.io/docs/secrets/kv/kv-v2.html).
```
Comment (bettyjunod) [inline docs/content/installation/gateway/docker-compose-file/_index.md]: Please call out as HashiCorp Consul

File: docs/content/installation/gateway/docker-compose-file/_index.md [hunk 3]
```diff
@@ -4,15 +4,160 @@ weight: 5
...
+
+{{% notice note %}}
+The deployment steps in this tutorial are for demonstration and learning on a local machine and should not be used in a production setting. Please refer to using either Kubernetes or Consul & Vault to provide storage of configuration and credentials.
+{{% /notice %}}
+
+---
+
+## Architecture
+
+Gloo Gateway without Kubernetes uses multiple pieces of software for deployment and functionality.
```
Comment (bettyjunod) [inline docs/content/installation/gateway/docker-compose-file/_index.md]: Change Gloo Gateway to just Gloo, unless you mean Gloo as API gateway

File: docs/content/installation/gateway/docker-compose-file/_index.md [hunk 4]
```diff
@@ -4,15 +4,160 @@ weight: 5
...
+
+- **Docker Compose**: The components of Gloo are deployed as containers running proxy, envoy, and the gateway
```
Comment (bettyjunod) [inline docs/content/installation/gateway/docker-compose-file/_index.md]: proxy, envoy and the gateway

is that gloo proxy? and then envoy proxy? can we spell out?

File: docs/content/installation/gateway/docker-compose-file/_index.md (hunk 1 above)
Comment (ned1313) [reply docs/content/installation/gateway/docker-compose-file/_index.md]: Updated.

File: docs/content/installation/gateway/docker-compose-file/_index.md (hunk 2 above)
Comment (ned1313) [reply docs/content/installation/gateway/docker-compose-file/_index.md]: Updated both Consul and Vault references.

File: docs/content/installation/gateway/docker-compose-file/_index.md (hunk 3 above)
Comment (ned1313) [reply docs/content/installation/gateway/docker-compose-file/_index.md]: Updated.

File: docs/content/installation/gateway/docker-compose-file/_index.md (hunk 4 above)
Comment (ned1313) [reply docs/content/installation/gateway/docker-compose-file/_index.md]: Reworded to Gloo, Envoy, and Gloo Gateway

**Comment by ned1313** [comment]
Changes made based on Betty's comments.

**Pull request merged**